pkg/crawler/crawler.go

```diff
@@ -117,56 +121,62 @@ loop:
 	return nil
 }
 
-func (c *Crawler) Visit(url string) error {
-	resp, err := c.http.Get(url)
-	if err != nil {
-		return xerrors.Errorf("http get error (%s): %w", url, err)
-	}
-	defer resp.Body.Close()
-
-	d, err := goquery.NewDocumentFromReader(resp.Body)
-	if err != nil {
-		return xerrors.Errorf("can't create new goquery doc: %w", err)
-	}
-
-	var children []string
-	var foundMetadata bool
-	d.Find("a").Each(func(i int, selection *goquery.Selection) {
-		link := selection.Text()
-		if link == "maven-metadata.xml" {
-			foundMetadata = true
-			return
-		} else if link == "../" || !strings.HasSuffix(link, "/") {
-			// only `../` and dirs have `/` suffix. We don't need to check other files.
-			return
+func (c *Crawler) Visit(ctx context.Context, url string) error {
+	select {
+	// Context can be canceled if we receive an error from another Visit function.
+	case <-ctx.Done():
+		return nil
+	default:
+		resp, err := c.http.Get(url)
```

Comment:
It is a good opportunity to make the HTTP request context-aware.

```suggestion
	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
	if err != nil {
		return xerrors.Errorf("unable to new HTTp request: %w", err)
	}

	if err = client.Do(req); err != nil {
```

Reply:
Good point!
Created this in https://github.com/aquasecurity/trivy-java-db/pull/27/commits/262f22df69044949a9a7961ed475cf987048e5cb.